lib/services/oauth2_service.ex

```diff
@@ -52,6 +52,49 @@ defmodule EaRestaurantDataLoader.Oauth2Service do
     }
   end
 
+  def refresh_token(refresh_token, access_token, client_id, client_secret) do
+    secret_key = Application.get_env(:ea_restaurant_data_loader, :secret_key)
+
+    client = get_client_by_client_id_and_entity_status(client_id, Status.active())
+
+    [client_scope | _] = client.scopes
+    validate_client_credentials(client, client_secret, Status.active())
+
+    case Oauth2Util.validate_token(access_token, secret_key) do
+      {:ok, _} ->
+        %{
+          :client_name => client.client_name,
+          :access_token => access_token,
+          :refresh_token => refresh_token,
+          :scopes => client_scope.scope,
+          :expire_in => client.access_token_expiration_time
+        }
+
+      {:error, _} ->
+        Oauth2Util.validate_token(refresh_token, secret_key)
```

Comment:
I think you should expect `{:ok,_}` from here in order to validate the token properly?

Reply:
Done.